str/eh_filter_extractor.py

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env python3
+
+"""
+This script extracts sample/locus combinations with EH binary filter set to "LowDepth" for all EH VCFs found within a directory into a .JSON format to be used as an annotation file in Hail query. 
+
+analysis-runner --access-level standard --dataset tob-wgs --description \
+    'EH Filter Extractor' --output-dir 'str/expansionhunter/v3-eh-filter-extractor' \
+    eh_filter_extractor.py \
+    --input-dir-eh=gs://cpg-tob-wgs-main/str/expansionhunter/v3 \
+    --output-name-eh=eh.json 
+"""
+
+import json
+import logging
+import click
+from cloudpathlib import GSPath
+from cyvcf2 import VCFReader
+import hailtop.batch as hb
+
+from cpg_utils import to_path
+from cpg_utils.config import get_config
+from cpg_utils.hail_batch import remote_tmpdir, output_path
+
+
+config = get_config()
+
+
+def eh_filter_extractor(input_dir):
+    """Creates a JSON file keyed by locus with value being a list of sample_ids with binary filter status == "LowDepth" in EH VCFs"""
+
+    files = to_path(input_dir).glob('*.vcf')
+    low_depth_dict = {}
+    for file in files:
+        if isinstance(file, GSPath):
+            file.copy(file.name)
+            file = file.name
+        else:
+            logging.warning(f'There is a file path that is not a GSPath: {file}')
+
+        reader = VCFReader(file)
+        sample_id = str(reader.samples[0])
+        for variant in reader:
+            locus = f'{variant.CHROM}:{variant.POS}'
+            e_qual = str(variant.FILTER)
+            if e_qual == 'LowDepth' and locus in low_depth_dict:
+                low_depth_dict[locus].append(sample_id)
+            elif e_qual == 'LowDepth' and locus not in low_depth_dict:
+                low_depth_dict[locus] = [sample_id]
```

Comment:
This will work fine, but there's a couple of python built-in functions that would make it easier:

```
from collections import defaultdict
low_depth_dict = defaultdict(list)
...
low_depth_dict[locus].append(sample_id)
```
a defaultdict can always be added to, even if the index didn't previously exist

Another option is [setdefault](https://docs.python.org/2/library/stdtypes.html#dict.setdefault), but that might not be as efficient... either get the current value, or create an entry if the key doesn't exist. In this example `low_depth_dict.setdefault(locus, [])` means 'get the value of low_depth_dict[locus], and if it doesn't exist, set low_depth_dict[locus] to a list, then get that'

```
low_depth_dict = {}
...
low_depth_dict.setdefault(locus, []).append(sample_id)
```